### ISSUE TITLE
Component doesn't update if Children resize

### ISSUE THREAD

**Issue by aaronhayes** [eiriklv/react-masonry-component#5]
The React-Masonry-Component does not update if a child component changes size - it causing overlapping issues/general ugliness. 
There should be a way to monitor children component for size changes.

For example in this scenario. If ChildOne's state is updated it will be re-rendered but the Masonry component would not be and therefore not realigned properly (unless it was listening for the same changes - which is not always practical). 

``` javascript
<Masonry options={options}>
   <ChildOne />
   <ChildTwo />
</Masonry 
```

Am I doing something wrong or is this an actual bug/not fixable?

Cheers!

might be the same issue as #2

**Comment by AugustinLF** [comment]
I had the same problem. As far as I can see, the only way is to change a props of Masonry (any, even if it doesn't exist), to force a reload (the component triggers a `reloadItems` on `ComponentWillReceiveProps`. Not perfect, but this is the same thing (IIRC) with Masonry in general (not about this component), even with vanilla JS, you still need to trigger the refresh.

**Comment by aaronhayes** [comment]
so my very hacky solution was to expose the performLayout function of React-Masonry and call it whenever I needed a refresh. It works but not very elegant.

**Comment by t1mmen** [comment]
I managed to work around this by simply setting state in the component that has `<Masonry />` in it's render method (didn't need to pass the state as prop, even)

**Comment by Narzerus** [comment]
@t1mmen's approach works, though this is clearly still a hacky approach.

**Comment by eiriklv** [comment]
@If anyone has a solution that keeps the existing functionality and fixes this then feel free to send a PR! This component is a bit hacky from the start actually, so a hacky solution that is encapsulated in the Masonry component (not leaking out) is better than no solution :-)

**Comment by ankur0890** [comment]
Facing the same issue. Unable to figure out a solution yet. Put this on stackoverflow also for more ideas http://stackoverflow.com/questions/34411211/reload-react-masonry-component-manually

**Comment by erkartik91** [comment]
I added reference to my Masonry instance using the following code
`ref={function(c) {this.masonry = c.masonry;}.bind(this)}`

and now can manually trigger resize using the following code
`this.masonry.layout()`

Hope this helps.

**Comment by afram** [comment]
fixed in `react-masonry-component@4.3.0`

**Comment by maurusrv** [comment]
hi, also encountered this issue and i used @erkartik91's solution and it worked. 

anyway, 
i am using this library: https://github.com/puikinsh/Adminator-admin-dashboard which uses react-masonry-component too as i've looked into my node_modules folder. according to the changelog, it's version is 6.2.0. 

last comment here says it's fixed even in the earlier version (4.3.0), but i wonder is it really fixed in the update? because i still encountered this same issue.

thank you.

**Comment by ldsmike88** [comment]
If anyone is still trying to figure this out, the new way to do this is:

`<Masonry enableResizableChildren>{ children }</Masonry>`

https://github.com/eiriklv/react-masonry-component/issues/116#issuecomment-438238947
